### PR TITLE
Update date.py 

### PR DESCRIPTION
**Pull request by mailhot001**
remove the + 1 on the week RE match in date parser so that the week number match the front end displayed week number.

related to: #1805

### CODE REVIEW

File: src/fava/util/date.py [hunk 1]
```diff
@@ -382,7 +382,7 @@ def parse_date(  # noqa: PLR0911
     match = WEEK_RE.match(string)
     if match:
         year, week = map(int, match.group(1, 2))
-        start = datetime.date.fromisocalendar(year, week + 1, 1)
```
Comment (yagebu) [inline src/fava/util/date.py:385]: this seems to have been my poor attempt at converting from "iso week number" to "%W-like week number".

I think we should switch to using `datetime.datetime.strptime(string, "%Y-W%W").date()`

**Review by mailhot001 — COMMENTED** [review]
@yagebu Here is my second try, I believe it matches the D3.js week parsing, I've tested it roughly (without edge case) and it is working the same as frontend.

File: src/fava/util/date.py (hunk 1 above)
Comment (mailhot001) [reply src/fava/util/date.py:385]: For your info, the tzinfo was suggested by tox, but is useless as Tz is not present in the date we extract afterwards.

**Review by yagebu — COMMENTED** [review]
Thanks :) Can you add some test cases that show that this is now working correctly in cases where it didn't before?

**Comment by mailhot001** [comment]
I added a test case to show an error if ISO week format is used. 
Also added a note in the help to explain the week numbering used, a month is a month, but week can be taken differently. The note should help users know how it work without checking the code and then checking the %W definition of strptime. 
Or the equivalent definition used in D3.js.

**Pull request merged**